internal/app/cli.go

```diff
@@ -221,49 +221,38 @@ func (c *cli) readState(s *state) {
 		if result {
 			log.Info(msg)
 		} else {
-			log.Fatal(msg)
+			return fmt.Errorf(msg)
 		}
 		// Merge Apps that already existed in the state
 		for appName, app := range fileState.Apps {
 			if _, ok := s.Apps[appName]; ok {
 				if err := mergo.Merge(s.Apps[appName], app, mergo.WithAppendSlice, mergo.WithOverride); err != nil {
-					log.Fatal("Failed to merge " + appName + " from desired state file" + f)
+					return fmt.Errorf("failed to merge %s from desired state file %s", appName, f)
 				}
 			}
 		}
 
 		// Merge the remaining Apps
 		if err := mergo.Merge(&s.Apps, &fileState.Apps); err != nil {
-			log.Fatal("Failed to merge desired state file" + f)
+			return fmt.Errorf("Failed to merge desired state file" + f)
 		}
 		// All the apps are already merged, make fileState.Apps empty to avoid conflicts in the final merge
 		fileState.Apps = make(map[string]*release)
 
 		if err := mergo.Merge(s, &fileState, mergo.WithAppendSlice, mergo.WithOverride); err != nil {
-			log.Fatal("Failed to merge desired state file" + f)
+			return fmt.Errorf("Failed to merge desired state file" + f)
 		}
 	}
 
-	s.setDefaults()
-	s.initializeNamespaces()
+	s.init() // Set defaults
 	s.disableUntargetedApps(c.group, c.target)
 
-	if len(c.target) > 0 && len(s.TargetMap) == 0 {
```

Comment:
In this PR, this validation now occurs later, right? Why?

Reply:
I've moved it because those cause helmsman to exit and I think it makes more sense to do that from main and I don't think it makes that much of a difference checking that before or after the syntax validation.
Now that I think about it, I think it's actually better to do it after so you learn about syntax errors in your DSF even if you made a mistake and targeted some app or group that doesn't exit, if you don't care about that you cant use `--skip-validation`...